packages/contracts/utils/ens.ts

```diff
@@ -46,15 +54,31 @@ export async function setupENS(domains: string[], hre: EHRE): Promise<any> {
         .filter(value => value !== '')
         .reverse()
         .join('.');
-      await ens.setSubnodeRecord(
+
+      // skipping if it is already set
+      const resolved = await ens.resolver(
```

Comment:
I don't get this check ! The above `if (resolved === resolver.address) {` is a great fix, but this one, I  couldn't imagine when this would be worth it.

Reply:
This one is necessary because the loop fails if we use a subdomain for one of the ENS configurations. e.g. `plugins.dao.eth`.
While registering `plugins.dao.eth` it would try to register `dao.eth` which already exists and fail